jax/_src/clusters/ompi_cluster.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2023 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import Optional
+from jax._src.clusters import ClusterEnv
+
+_ORTE_URI = 'OMPI_MCA_orte_hnp_uri'
+_PROCESS_COUNT = 'OMPI_COMM_WORLD_SIZE'
+_PROCESS_ID = 'OMPI_COMM_WORLD_RANK'
+_LOCAL_PROCESS_ID = 'OMPI_COMM_WORLD_LOCAL_RANK'
+
+class OmpiCluster(ClusterEnv):
+  @classmethod
+  def is_env_present(cls) -> bool:
+    return _ORTE_URI in os.environ
+
+  @classmethod
+  def get_coordinator_address(cls) -> str:
+    # Example of orte_uri:
+    # 1531576320.0;tcp://10.96.0.1,10.148.0.1,10.108.0.1:34911
+    orte_uri = os.environ[_ORTE_URI]
+    job_id = orte_uri.split('.', maxsplit=1)[0][:5]
+    # Pick port in ephemeral range [(65535 - 2^12 + 1), 65535]
+    port = int(job_id) % 2**12 + (65535 - 2**12 + 1)
+    launcher_ip = orte_uri.split('//', maxsplit=1)[1]
+    launcher_ip = launcher_ip.split(':', maxsplit=1)[0]
+    launcher_ip = launcher_ip.split(',', maxsplit=1)[0]
```

Comment:
A regexp might be more readable but hard to say :)
`launcher_ip = re.search("tcp://([0-9.]+),", orte_uri).group(1)`
or
`launcher_ip = re.search("tcp://(.+?),", orte_uri).group(1)`